src/util.rs

```diff
@@ -1,49 +1,59 @@
 use std::hash::{Hasher, Hash};
-use ::byteorder::{BigEndian, WriteBytesExt, ReadBytesExt};
+use ::byteorder::{BigEndian, WriteBytesExt};
 use ::bucket::{Fingerprint, FINGERPRINT_SIZE};
 
+// A struct combining *F*ingerprint *a*nd *I*ndexes,
+// to have a return type with named fields
+// instead of a tuple with unnamed fields.
 pub struct FaI {
     pub fp: Fingerprint,
     pub i1: usize,
     pub i2: usize,
 }
 
-fn get_hash<T: ?Sized + Hash, H: Hasher + Default>(data: &T) -> [u8; 4] {
-    let mut result = [0; 4];
+fn get_hash<T: ?Sized + Hash, H: Hasher + Default>(data: &T) -> (u32, u32) {
+    let result: u64;
+    // TODO: why the additional scope?
```

Comment:
@seiflotfy could you explain the reasoning for this additional scope inside the function to me?

Reply:
I am confused myself why I did that... there is no need for it

Reply:
Ok, I thought maybe the Hasher needs to be dropped early… But I could not find something like that and my integration test works without it as well, so cleaned up in 824bced21e8ffc0eb6cdcdbf45a90068a3da3b42. ✨